jit-compiler/src/codegen.rs

```diff
@@ -436,7 +441,7 @@ fn format_signed_integer(n: &BigInt) -> String {
     if let Ok(n) = BigUint::try_from(n) {
         format_unsigned_integer(&n)
     } else {
-        format_signed_integer(&-(n.clone()))
+        format!("-{}", format_signed_integer(&-(n.clone())))
```

Comment:
Is clone really necessary?

Anyway, maybe it is better to avoid recursion:
```suggestion
        format!("-{}", format_unsigned_integer(&BigUint::try_from(-n).unwrap())
```